### PR TITLE
ipc improvements, fix slow shutdown & cleanup leaked resources

### PR DESCRIPTION
**Pull request by theomonnom**
- fix zombie processes when the main process doesn't gracefully exit
- fix slow shutdown
- fix some unclosed resources
- fix some leaked mp processes (pipe & Queue)
- add a moving avg to the default cpu_load_fnc
- added async duplex impl using socket.sockerpair

### CODE REVIEW

**Review by davidzhao — APPROVED** [review]

**Comment by theomonnom** [comment]
cc @egoldschmidt

**Review by davidzhao — APPROVED** [review]
lg

**Comment by theomonnom** [comment]
going to use socketpair to create asynchronous/non-blocking IPC communication before merging this PR

File: livekit-agents/livekit/agents/ipc/proc_pool.py [hunk 1]
```diff
@@ -83,8 +83,9 @@ async def _proc_watch_task(self) -> None:
                 # neither be used to launch jobs
                 self.emit("process_ready", proc)
                 self._warmed_proc_queue.put_nowait(proc)
-            except asyncio.TimeoutError:
+            except Exception:
                 self._proc_needed_sem.release()  # notify to warm a new process after initialization failure
+                pass
```
Comment (keepingitneil) [inline livekit-agents/livekit/agents/ipc/proc_pool.py:88]: remove

**Review by keepingitneil — APPROVED** [review]

**Pull request merged**